test-project/helloFX/build.gradle

```diff
@@ -49,9 +45,11 @@ task runScript(type: Exec) {
     def argsList = ["${java_home}/bin/java", '-cp', substratePath,
                     "-Dimagecp=${javafx_path}/javafx.base.jar:${javafx_path}/javafx.graphics.jar:${javafx_path}/javafx.controls.jar:" +
                             "${project.sourceSets.main.runtimeClasspath.filter{!it.name.contains('javafx')}.asPath}",
-                    "-Dgraalvm=$graal_path", "-Dprism.sw=true"]
+                    "-Dgraalvm=$graal_path"]
     if (target != null) {
         argsList += ["-DtargetProfile=$target"]
+    } else {
+        argsList += ["-Dprism.sw=true"]
```

Comment:
if target == null sw rendering needs to be used? 
That sounds unrelated to each other.

Reply:
This is for running HelloFX on travis/jenkins without a display. Since we don't run on iOS (target != null), it is an easy way to add that condition.

Reply:
That is very brittle. 
What you want to say is that if the "target" is "ios", you want to specify -DtargetProfile=ios.
If the target is not ios (e.g. it is not set, or it is "linux" or it is "armv6hf" or so, you need software rendering.

But I think what you really want to say is that in case of crosscompilation, don't bother about displaying.
In case of crosscompilation, specify the "target" as a profile.